### PR TITLE
Test framework: only add the test jar as additional archive if it's different from the app jar

### LINKED ISSUE #3615: Test framework: only add the test jar as additional archive if it's different from the app jar
Currently, when initializing the RuntimeRunner in the QuarkusTestExtension we assume the test location and the app location are different (in case the tests are provided in a JAR). This could not be the case. In which case, we shouldn't be adding the JAR containing the tests as an additional archive to be indexed. Otherwise the same location gets indexed twice which leads to build errors.

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes #3615

### CODE REVIEW

**Review by geoand — APPROVED** [review]
:+1:

File: devtools/gradle/gradle.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+version = 999-SNAPSHOT
```
Comment (gsmet) [inline devtools/gradle/gradle.properties]: Hmm, is it related to the issue at hand?

This has consequences on the release process if we add a version in the middle of things.

**Review by gsmet — DISMISSED** [review]
Please see my question inline.

File: devtools/gradle/gradle.properties (hunk 1 above)
Comment (geoand) [reply devtools/gradle/gradle.properties]: Oh, I didn't know that, sorry

Comment (aloubyansky) [reply devtools/gradle/gradle.properties]: Ah, sorry, good catch. It's not related. I keep adding this file locally when I'm working on gradle issues.

Comment (aloubyansky) [reply devtools/gradle/gradle.properties]: Removed it.

**Pull request merged**